github/resource_github_repository_milestone.go

```diff
@@ -0,0 +1,232 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v31/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func resourceGithubRepositoryMilestone() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryMilestoneCreate,
+		Read:   resourceGithubRepositoryMilestoneRead,
+		Update: resourceGithubRepositoryMilestoneUpdate,
+		Delete: resourceGithubRepositoryMilestoneDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), "/")
+				if len(parts) != 3 || parts[0] == "" || parts[1] == "" || parts[2] == "" {
+					return nil, fmt.Errorf("Invalid ID format, must be provided as OWNER/REPOSITORY/NUMBER")
+				}
+				d.Set("owner", parts[0])
+				d.Set("repository", parts[1])
+				number, err := strconv.Atoi(parts[2])
+				if err != nil {
+					return nil, err
+				}
+				d.Set("number", number)
+				d.SetId(fmt.Sprintf("%s/%s/%d", parts[0], parts[1], number))
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"title": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"due_date": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "in yyyy-mm-dd format",
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"open", "closed",
+				}, true),
+				Default: "open",
+			},
+			"number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+const (
+	layoutISO = "2006-01-02"
+)
+
+func resourceGithubRepositoryMilestoneCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*Organization).v3client
+	ctx := context.Background()
```

Comment:
Recently came across this:

https://github.com/terraform-providers/terraform-provider-github/blob/518829b4f87f8613feaa86b4046702a962619830/github/provider.go#L136

Does the provider become more responsive to a cancel operation if we wire this context all the way through to requests like this?